apps/wallet-mobile/src/features/SetupWallet/useCases/RestoreWallet/RestoreWalletScreen.tsx

```diff
@@ -110,34 +105,13 @@ export const RestoreWalletScreen = () => {
     }, [mnemonicType, track]),
   )
 
-  const handleOpenWalletWithDuplicatedName = React.useCallback(
-    (walletMeta: Wallet.Meta) => {
-      walletManager.setSelectedWalletId(walletMeta.id)
-      resetToTxHistory()
-    },
-    [walletManager, resetToTxHistory],
-  )
-
   const handleOnNext = React.useCallback(async () => {
-    const {csl, release} = wrappedCsl()
-
-    const {accountPubKeyHex} = await keyManager(walletImplementation)({mnemonic, csl, accountVisual})
-    const checksum = walletChecksum(accountPubKeyHex)
-    release()
-
-    const duplicatedWalletMeta = Array.from(walletManager.walletMetas.values()).find(
-      (walletMeta) => walletMeta.plate === checksum.TextPart,
-    )
-
-    if (duplicatedWalletMeta) {
-      openModal(
-        strings.restoreDuplicatedWalletModalTitle,
-        <Modal
-          walletName={duplicatedWalletMeta.name}
-          publicKeyHex={accountPubKeyHex}
-          onPress={() => handleOpenWalletWithDuplicatedName(duplicatedWalletMeta)}
-        />,
-      )
+    const {accountPubKeyHex} = await walletManager.generateWalletKeys(walletImplementation, mnemonic, accountVisual)
+
+    const iswalletDuplicated = walletManager.isWalletAccountDuplicated(accountPubKeyHex)
```

Comment:
```suggestion
    const isWalletDuplicated = walletManager.isWalletAccountDuplicated(accountPubKeyHex)
```

Reply:
this might take awhile, for now let's leave it, but eventually it will require a pending state.